format-common/src/main/java/io/cdap/plugin/format/connector/AbstractFileConnector.java

```diff
@@ -0,0 +1,186 @@
+      FileInputFormat.addInputPath(job, path);
+      return null;
+    });
+
+    String inputFormatClassName = inputFormat.getInputFormatClassName();
+    Configuration hConf = job.getConfiguration();
+    Map<String, String> inputFormatConfiguration = inputFormat.getInputFormatConfiguration();
+    for (Map.Entry<String, String> propertyEntry : inputFormatConfiguration.entrySet()) {
+      hConf.set(propertyEntry.getKey(), propertyEntry.getValue());
+    }
+
+    // set entries here again, in case anything set by PathTrackingInputFormat should be overridden
+    for (Map.Entry<String, String> entry : getFileSystemProperties(fullPath).entrySet()) {
+      conf.set(entry.getKey(), entry.getValue());
+    }
+
+    return new SourceInputFormatProvider(inputFormatClassName, conf);
+  }
+
+  @Override
+  public void test(ConnectorContext connectorContext) throws ValidationException {
+    // no-op
+  }
+
+  @Override
+  public StructuredRecord transform(NullWritable key, StructuredRecord val) {
+    return val;
+  }
+
+  @Override
+  public ConnectorSpec generateSpec(ConnectorContext context,
+                                    ConnectorSpecRequest connectorSpecRequest) throws IOException {
+    ConnectorSpec.Builder builder = ConnectorSpec.builder();
+
+    String path = getFullPath(connectorSpecRequest.getPath());
+    ValidatingInputFormat inputFormat = getValidatingInputFormat(context, path);
+    // TODO: CDAP-18060 in 6.5 this will only have text and blob that will not access the file system,
+    //  to support other formats, we need to ensure get schema works
+    Schema schema = inputFormat.getSchema(new FormatContext(context.getFailureCollector(), null));
+    builder.setSchema(schema);
+    setConnectorSpec(connectorSpecRequest, builder);
+    return builder.build();
+  }
+
+  /**
+   * Return the full path including the scheme for the connector
+   */
+  protected String getFullPath(String path) {
+    return path;
+  }
+
+  /**
+   * Override this to provide any additional Configuration properties that are required by the FileSystem.
+   * For example, if the FileSystem requires setting properties for credentials, those should be returned by
+   * this method.
+   *
+   * @param path the path for the file system, some file system needs this path, for example, gcs needs the path to
+   *             determine the bucket name
+   */
+  protected Map<String, String> getFileSystemProperties(String path) {
+    return Collections.emptyMap();
+  }
+
+  /**
+   * Override this method to provide related plugins, properties or schema
+   *
+   * @param request the spec generation request
+   * @param builder the builder of the spec
+   */
+  protected void setConnectorSpec(ConnectorSpecRequest request, ConnectorSpec.Builder builder) {
+    // no-op
+  }
+
+  private ValidatingInputFormat getValidatingInputFormat(ConnectorContext context, String path) throws IOException {
+    String fileType = FileTypeDetector.detectFileType(path);
```

Comment:
we said directories are sample-able earlier, what does this do for directories? It seems like it returns that it's text?

Reply:
Yes this will return text for it. UI now won't lead users to sample a directory since it is browsable. This will work much better once we let user input the format and UI honor users to sample a directory